src/main/java/com/aws/greengrass/mqtt/bridge/clients/IoTCoreClient.java

```diff
@@ -244,12 +250,16 @@ private void subscribeToTopicsWithRetry(Set<String> topics) {
 
     private void subscribeToIotCore(String topic)
```

Comment:
why isn't `nolocal` being passed in here? (or rather an object that represents the route)

Reply:
It can be, but we have that info in routesByTopic so we can just look it up without needing the param.  I don't feel strongly about it though 🤷

Reply:
Routes by topic is something you just added here right?

I'm seeing 19 files changed when it _feels_ like this should be 3 files plus a test

Reply:
Yeah it took some refactoring to be able to pass route config into IoTCoreClient because we need to read bridge config only once it's available.  This also touches both IoTCoreClient and LocalMqtt5Client, their tests and changes to integration test extension (because of aforementioned refactoring) Touches a lot of files but I think it's necessary. Could also split into multiple PRs